sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkLabelSet.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import io.opentelemetry.metrics.LabelSet;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@AutoValue
+abstract class SdkLabelSet implements LabelSet {
+
+  private static final LabelSet EMPTY = create();
+
+  public static LabelSet empty() {
+    return EMPTY;
+  }
+
+  public static LabelSet create(String... keyValuePairs) {
```

Comment:
I think this can be package protected, apply same everywhere.

Reply:
Probably makes sense to change the API to accept Map, and do few optimizations:
1) If map is empty return empty.
2) If map has 1/2 elements create special autovalue classes for this.
3) Otherwise use this generic class.

Reply:
Are you thinking that the "special autovalue" instances would have a different api, or should they all have a uniform `Map<String, String> getLabels()` method on them, just backed with different data?

Reply:
Also, I'd definitely rather worry about this kind of micro-optimizations until after we've actually got a metric pipeline built and being exercised, and we've benchmarked and found this to be a problem.

Reply:
Yes, we can just do the 1 and 3 and avoid optimizations.

Reply:
done